server/services/transaction.service.js

```diff
@@ -146,3 +149,21 @@ export const getTotalAmountPaidByUser = async (userId) =>
       },
     },
   ]);
+
+export const getReferralRewards = async (userId) =>
+  Offer.aggregate([
+    { $match: { userId: ObjectId(userId) } },
+    {
+      $match: {
+        $or: [
+          { status: OFFER_STATUS.GENERATED },
+          { status: OFFER_STATUS.INTERESTED },
+          { status: OFFER_STATUS.ASSIGNED },
+          { status: OFFER_STATUS.ALLOCATED },
+        ],
+      },
+    },
+  ]);
+
+export const getContributionRewards = async (referrerId) =>
```

Comment:
It looks like you mixed this up, this is getting Referrals not contribution rewards 😏 

```suggestion
export const getReferralRewards = async (referrerId) =>
```